dvc/fs/hdfs.py

```diff
@@ -220,8 +221,19 @@ def isdir(self, path_info):
 
     def info(self, path_info):
         with self.hdfs(path_info) as hdfs:
-            finfo = hdfs.get_file_info(path_info.path)
-            return {"size": finfo.size}
+            from pyarrow.fs import FileType
+
+            file_info = hdfs.get_file_info(path_info.path)
+            if file_info.type is FileType.Directory:
+                kind = "directory"
+            elif file_info.type is FileType.File:
+                kind = "file"
+            else:
+                raise FileNotFoundError(
+                    errno.ENOENT, os.strerror(errno.ENOENT), path_info
+                )
```

Comment:
This `else` is not obvious, could you elaborate why we raise this if the type doesn't match dir or file?

Reply:
E.g. if you look at fsspec's hdfs: https://github.com/intake/filesystem_spec/blob/9c75d6267d930b4fb01333d22bc2bebf4836116c/fsspec/implementations/hdfs.py#L105

Reply:
`get_file_info()` won't raise any files even if the `path_info` does not exists. It would return `FileType.NotFound` (or `FileType.Unknown` but I don't think that is a case that we need to handle separately).

Reply:
@isidentical Ah, ok, makes sense. :+1: